src/defs.jl

```diff
@@ -0,0 +1,7 @@
+# Constants
+
+@enum Variant begin
```

Comment:
For the variant, better use a struct than an enum, it lets anyone else extend to a new algorithm when they need/want